osquery/dispatcher/scheduler.h

```diff
@@ -43,6 +44,45 @@ class SchedulerRunner : public InternalRunnable {
   std::chrono::milliseconds getCurrentTimeDrift() const noexcept;
 
  private:
+   inline void calculateTimeDrift(std::chrono::time_point<std::chrono::steady_clock> start_time_point) {
+     auto loop_step_duration =
+         std::chrono::duration_cast<std::chrono::milliseconds>(
+             std::chrono::steady_clock::now() - start_time_point);
+     if (loop_step_duration + time_drift_ < interval_) {
+       pauseMilli(interval_ - loop_step_duration - time_drift_);
+       time_drift_ = std::chrono::milliseconds::zero();
+     } else {
+       time_drift_ += loop_step_duration - interval_;
+       if (time_drift_ > max_time_drift_) {
+         // giving up
+         time_drift_ = std::chrono::milliseconds::zero();
+       }
+     }
+   }
+
+   inline void maybeRunDecorators(size_t i) {
```

Comment:
Variable name `i` used to make sense in context of loop, but here I'd suggest to rename it to something more expressive. e.g.: `time_step` or `step_in_seconds`.